src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs

```diff
@@ -13,7 +15,15 @@ public abstract class JsonConverter<T> : JsonConverter
         /// <summary>
         /// When overidden, constructs a new <see cref="JsonConverter{T}"/> instance.
         /// </summary>
-        protected internal JsonConverter() { }
+        protected internal JsonConverter()
+        {
+            // Today only typeof(object) can have polymorphic writes.
+            // In the future, this will be check for !IsSealed (and excluding value types).
+            CanBePolymorphic = TypeToConvert == typeof(object);
+
+            IsInternalConverter = GetType().Assembly == typeof(JsonConverter).Assembly;
```

Comment:
Couldn't we set `IsInternalConverter` as `true` on the most derived types so we can recognize them as built-in instead of comparing assemblies?

Reply:
Yes. We could set all all of these on each converter but it's done here to avoid any inconsistencies and easier maintainability. Since the converters are cached, this is only done once per Type.